be/src/main/java/com/savethepet/config/SecurityConfig.java

```diff
@@ -0,0 +1,91 @@
+package com.savethepet.config;
+
+import com.savethepet.service.CustomUserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.oauth2.config.annotation.web.configuration.EnableOAuth2Client;
+
+
+/**
+ * Configuration of Spring Security: *
+ * 1) Basic auth
+ * 2) Google Oauth2
+ * 3) Yandex Oauth2
+ * 4) Vk Oauth2
+ * @author Alexey Klimov
+ */
+@Configuration
+@EnableWebSecurity
+@EnableOAuth2Client
+public class SecurityConfig extends WebSecurityConfigurerAdapter {
+
+
+    @Autowired
+    private CustomUserService customUserService;
+
+
+    @Bean
+    PasswordEncoder passwordEncoder() {
+        return new BCryptPasswordEncoder();
+    }
+
+
+    /**
+     * Configure for http requests
+     *
+     * @param http
+     * @throws Exception
+     */
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http
+                .csrf()
+                    .disable()
+                .authorizeRequests()
+                    .antMatchers("/login",
+                            "/registration",
+                            "/home",
```

Comment:
I think it's a bad idea to use the endpoint to save a user with a not-registered user, but it would be nice to change it to `/welcome`, because the page with information is available to everyone